plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -188,6 +193,44 @@ def __cmp__(self, other):
             other.complete_version_serialized
         )
 
+    def get_or_calculate_checksum(self, checksumtype):
+        """
+        Tries to get the sha1 or sha256 checksum value from the attribute on self of the
+        corresponding name. If it is not available, calculates the value and saves it on self
+        before returning it.
+
+        SIDE EFFECT: This method will save the model if the requested checksumtype is not populated
+                     on the model in the "checksums" attribute.
+
+        :param checksumtype:   any checksum type supported by the platform
+        :type  checksumtype:   basestring
+
+        :return:    checksum value of the requested type
+        :rtype:     basestring
+
+        :raises PulpCodedException: if the checksumtype is not available
+        """
+        if checksumtype not in server_util.CHECKSUM_FUNCTIONS:
+            raise ValueError(_('Checksum type %(checksumtype)s is not supported'))
```

Comment:
I think coded exceptions are better because the exception being coded can be talked about in a support discussion by number. It's a very small improvement so whatever you decide here is fine with me.